### DISCUSSION TITLE
How to decode qscc GetChainInfo using fabric-gateway?

### DISCUSSION THREAD

**Discussion by bh4rtp2023** [hyperledger/fabric-gateway#610] [Q&A] [answered]
```

const { common: protosCommon } = require('@hyperledger/fabric-protos');
...
const contract = gateway.getNetwork('mychannel').getContract('qscc');
const result = await contract.evaluateTransaction('GetChainInfo', 'mychannel');
const blockProto = JSON.stringify(protosCommon.BlockchainInfo.decode(result));
console.log(`GetChainInfo: ${blockProto} `);

```

It fails with error: `protosCommon.BlockchainInfo.decode is not a function`
How to use fabric-protos to decode the result of `GetChainInfo`? 
Thank you.

**Comment by bestbeforetoday** [comment, marked as answer]
There is a brief usage example in the [README for the @hyperledger/fabric-protos](https://www.npmjs.com/package/@hyperledger/fabric-protos) package. Since the protobuf bindings include full TypeScript type definitions, a good IDE _should_ be proving some auto-completion to help you too. The method you want is `deserializeBinary()`:

```typescript
const blockInfo = common.BlockchainInfo.deserializeBinary(result);
```

**Comment by bh4rtp2023** [comment]
Thank you @bestbeforetoday , I switched to use `fabric-protos` from `fabric-node-sdk`. Your answer simplifies the solution:)
